addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/change/MetadataMergePomChangeListener.java

```diff
@@ -0,0 +1,175 @@
+import org.commonjava.indy.pkg.maven.content.cache.MavenVersionMetadataCache;
+import org.commonjava.indy.pkg.maven.content.group.MavenMetadataMerger;
+import org.commonjava.indy.subsys.infinispan.CacheHandle;
+import org.commonjava.maven.galley.event.EventMetadata;
+import org.commonjava.maven.galley.event.FileAccessEvent;
+import org.commonjava.maven.galley.event.FileDeletionEvent;
+import org.commonjava.maven.galley.event.FileEvent;
+import org.commonjava.maven.galley.model.Transfer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executor;
+
+import static org.commonjava.indy.model.core.StoreType.hosted;
+import static org.commonjava.indy.util.LocationUtils.getKey;
+import static org.commonjava.maven.galley.util.PathUtils.normalize;
+import static org.commonjava.maven.galley.util.PathUtils.parentPath;
+
+@javax.enterprise.context.ApplicationScoped
+public class MetadataMergePomChangeListener
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    private StoreDataManager dataManager;
+
+    @Inject
+    private DownloadManager fileManager;
+
+    @Inject
+    private IndyFileEventManager fileEvent;
+
+    @Inject
+    @MavenVersionMetadataCache
+    private CacheHandle<StoreKey, Map> versionMetadataCache;
+
+    @Inject
+    @WeftManaged
+    @ExecutorConfig( daemon = true, priority = 7, named = "indy-events" )
+    private Executor executor;
+
+    public void metaClear( @Observes final FileEvent event )
+    {
+
+        final String path = event.getTransfer().getPath();
+
+        if ( event instanceof FileAccessEvent )
+        {
+            return;
+        }
+
+        if ( !path.endsWith( ".pom" ) )
+        {
+            return;
+        }
+
+        final StoreKey key = getKey( event );
+        final String versionPath = normalize( parentPath( path ) );
+        final String clearPath = normalize( normalize( parentPath( versionPath ) ), MavenMetadataMerger.METADATA_NAME );
+        try
+        {
+            if ( hosted == key.getType() )
+            {
+                HostedRepository hosted = dataManager.query().getHostedRepository( key.getName() );
+                try
+                {
+                    if ( doClear( hosted, clearPath ) )
+                    {
+                        versionMetadataCache.remove( hosted.getKey() );
+                    }
+                }
+                catch ( final IOException e )
+                {
+                    logger.error(
+                            String.format( "Failed to delete: %s from hosted: %s when pom version changed. Error: %s",
+                                           path, hosted, e.getMessage() ), e );
+                }
+
+                final Set<Group> groups = dataManager.query().getGroupsContaining( key );
```

Comment:
I think you might want getAffectedGroups() instead of just the groups containing the store. Groups of groups may also have cached metadata that is affected by this change.